android/build.gradle

```diff
@@ -1,56 +1,57 @@
 buildscript {
-  repositories {
-    jcenter()
-    google()
-    maven {
-      url 'https://maven.google.com'
+    repositories {
+        jcenter()
+        maven {
+            url 'https://maven.google.com/'
+            name 'Google'
+        }
     }
-  }
 
-  dependencies {
-    classpath 'com.android.tools.build:gradle:3.0.0'
-  }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:+'
+    }
 }
 
 apply plugin: 'com.android.library'
 
-def DEFAULT_COMPILE_SDK_VERSION             = 26
-def DEFAULT_BUILD_TOOLS_VERSION             = "26.0.2"
-def DEFAULT_TARGET_SDK_VERSION              = 26
-def DEFAULT_GOOGLE_PLAY_SERVICES_VERSION    = "12.0.1"
-def DEFAULT_SUPPORT_LIBRARY_VERSION         = "27.1.0"
+def DEFAULT_COMPILE_SDK_VERSION = 26
+def DEFAULT_BUILD_TOOLS_VERSION = "26.0.2"
+def DEFAULT_TARGET_SDK_VERSION = 26
+def DEFAULT_GOOGLE_PLAY_SERVICES_VERSION = "12.0.1"
+def DEFAULT_SUPPORT_LIBRARY_VERSION = "27.1.0"
 
 android {
-  compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
-  buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
-
-  defaultConfig {
-    minSdkVersion 16
-    targetSdkVersion rootProject.hasProperty('targetSdkVersion') ? rootProject.targetSdkVersion : DEFAULT_TARGET_SDK_VERSION
-
-    versionCode 1
-    versionName "1.0.0"
-  }
-  lintOptions {
-    abortOnError false
-    warning 'InvalidPackage'
-  }
+    compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
+    buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
+
+    defaultConfig {
+        minSdkVersion 16
+        targetSdkVersion rootProject.hasProperty('targetSdkVersion') ? rootProject.targetSdkVersion : DEFAULT_TARGET_SDK_VERSION
+
+        versionCode 1
+        versionName "1.0.0"
+    }
+    lintOptions {
+        abortOnError false
+        warning 'InvalidPackage'
+    }
 }
 
 repositories {
-  mavenCentral()
-  maven {
-   url 'https://maven.google.com'
-  }
-  maven {
-    // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
-    url "$rootDir/../node_modules/react-native/android"
-  }
+    mavenCentral()
+    maven {
+        url 'https://maven.google.com/'
+        name 'Google'
+    }
+    maven {
+        // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+        url "$rootDir/../node_modules/react-native/android"
+    }
 }
 
 dependencies {
-    def supportLibVersion = rootProject.hasProperty('supportLibVersion')  ? rootProject.supportLibVersion : DEFAULT_SUPPORT_LIBRARY_VERSION
+    def supportLibVersion = rootProject.hasProperty('supportLibVersion') ? rootProject.supportLibVersion : DEFAULT_SUPPORT_LIBRARY_VERSION
 
-    compileOnly 'com.facebook.react:react-native:+'
```

Comment:
We should keep `compileOnly`.
We can experience some conflicts with other apps.

Reply:
I tried to use compileOnly, but in projects using older gradle tools, they give error during compilation. This was happening in one of our projects.  

Following was the error:
```
Could not find method compileOnly() for arguments [com.facebook.react:react-native:+] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.
```

Reply:
Alright, but let's keep in mind that we should not support older gradle tools in future.
@sibelius, @jgfidelis and me also supports this repo: https://github.com/react-native-community/react-native-camera
We've been through great experiences with native modules and supporting lot's of versions.

My point is, let's test those new approaches you're proposing, since this changes are looking great, but we should follow Google and Apple best practices.

Thanks anyway \o/
I am very happy that someone is trying to get this package healthy.